compiler/passes/docs.cpp

```diff
@@ -295,6 +295,76 @@ std::string generateSphinxProject(std::string dirpath) {
   return std::string(moddir);
 }
 
+/* Checks for the valid Project Version (non-recursively). If an error occurs, exit and report
+ * error.
+ */
+static char * checkProjectVersion(char * projectVersion) {
+  int length = strlen(projectVersion);
+  int i = 0;
+  int dot = 0;
+  bool check = true;
+  int tagIndex = 0;
+
+  // Supported version tags
+  const char * tags[] = {"alpha", "beta", "rc"};
+  const char * error = "";
+  for(i=0; i<length; i++) {
+    if(i>0 && projectVersion[i] == '.') {
+      if(projectVersion[i-1] != '.') {
+        dot++;
+        if(dot>2) {
+          error = "Required only two dots which seperates three numbers";
+          check = false;
+          break;
+        }
+        if(i == length-1) {
+          error = "Cannot end with dot, can end with either number or tag";
+          check = false;
+          break;
+        }
+      } else {
+        error = "Missing number between dots";
+        check = false;
+        break;
+      }
+    } else if(projectVersion[i] == '-' && dot == 2) {
+      if(projectVersion[i-1] != '.') {
+        tagIndex = i+1;
+        break;
+      } else {
+        error = "Missing number before tag";
+        check = false;
+        break;
+      }
+    } else if(projectVersion[i] == '-' && dot != 2) {
+      error = "Required only two dots which seperates three numbers";
+      check = false;
+      break;
+    } else if((int)projectVersion[i] > (int)'9' || (int)projectVersion[i] < (int)'0') {
+      error = "Invalid Characters, only digits, dot and hyphen permitted";
```

Comment:
Technically this is incorrect.  We permit other characters, but only after the hyphen and only if they match our expected tags.  I would change this error message to "... only digits and dots permitted before a hyphen" or something along those lines